lib/chef/formatters/error_inspectors/compile_error_inspector.rb

```diff
@@ -44,6 +44,18 @@ def add_explanation(error_description)
             error_description.section("Cookbook Trace:", traceback)
             error_description.section("Relevant File Content:", context)
           end
+
+          if exception_message_modifying_frozen?
+            msg = <<-MESSAGE
+            Chef calls the freeze method on certain ruby objects to prevent
+            pollution across multiple instances. Specifically, resource
+            properties have frozen default values to avoid modifying the
+            property for all instances of a resource. Try modifying the
+            particular instance variable or using an instance accessor instead.
```

Comment:
Can this be made more clear? Should we have code examples? Do we need to mention `dup`? Also, values from `ENV` can be frozen, should we mention this?